File: 06-prime-check/prime_check.erl
%% Erlang program to check if a number is prime or not
%% Uses the O(sqrt(n)) algorithm
-module(prime_check).
-author("M.V.Harish Kumar").
-compile([{date, "2023-12-07"}]).
-export([main/0]).

is_prime(2) -> true;
is_prime(1) -> false;
is_prime(N) when N > 1 ->
    Sqrt = trunc(math:sqrt(N)),
    Pred = fun(I) -> N rem I == 0 end,
    not lists:any(Pred, lists:seq(2, Sqrt)).

main() ->
    io:format("Prime numbers Checker~n~n"),
    {ok, [Num]} = io:fread("Enter the number to check for prime: ", "~d"),
    case is_prime(Num) of
        true -> io:format("~w is Prime.", [Num]);
        false -> io:format("~w is NOT Prime.", [Num])
    end,
    erlang:halt(0).

File: 06-prime-check/PrimeCheck.cs
// C# program to check if a given number is prime or not
// Created by M.V.Harish Kumar on 06-01-2024
using System;

namespace PrimeCheck
{
    class PrimeCheck
    {
        static bool IsPrime(int n) {
            if (n == 2)
                return true;
            
            if (n < 1)
                return false;
            
            int ulim = Convert.ToInt32(Math.Sqrt(n));
            for (int i = 2; i <= ulim; i++)
            {
                if (n % i == 0)
                    return false;
            }

            return true;
        }

        static void Main(string[] args)
        {
            Console.Write("Enter a number: ");
            int num = Convert.ToInt32(Console.ReadLine());
            if (IsPrime(num))
            {
                Console.WriteLine($"{num} is Prime");
            }
            else
            {
                Console.WriteLine($"{num} is NOT Prime");
            }
        }
    }
}